engine/class_modules/sc_death_knight.cpp

```diff
@@ -9449,6 +9490,10 @@ void death_knight_t::create_buffs()
   buffs.frenzied_monstrosity = make_buff( this, "frenzied_monstrosity", find_spell ( 334896 ) )
     -> add_invalidate( CACHE_ATTACK_SPEED )
     -> add_invalidate( CACHE_PLAYER_DAMAGE_MULTIPLIER );
+
+  buffs.death_turf = make_buff( this, "death_turf", find_spell ( 335180) )
+    -> set_default_value( specialization() == DEATH_KNIGHT_BLOOD ? find_spell( 335180 ) -> effectN(1).percent() / 2 : find_spell( 335180 ) -> effectN(1).percent() )
+    -> add_invalidate( CACHE_HASTE );
```

Comment:
You can use set_pct_buff_type() so you don't have to add the buff to the player's composite melee/spell haste methods manually. 
Default value could be set to that of effect#1, and halved for blood after that, the end result would look like:
```
buffs.death_turf = make_buff( this, "death_turf", find_spell ( 335180 ) )
    -> set_default_value_from_effect( 1 )
    -> set_pct_buff_type( STAT_PCT_BUFF_HASTE );
if ( specialization() == DEATH_KNIGHT_BLOOD )
  buffs.death_turf -> default_value /= 2;
```